### PR TITLE
Additions of correlation type reading and writing

### PR DESCRIPTION
**Pull request by paulmillr**
Basically write support.

Closes gh-29. Please merge.

### CODE REVIEW

**Comment by EddyVerbruggen** [comment]
Thanks @paulmillr - gonna test this a bit now.

File: src/ios/HealthKit.m [hunk 1]
```diff
@@ -796,4 +953,97 @@ - (HKSampleType*) getHKSampleType:(NSString*) elem {
...
+    }
+    NSDictionary *metadata = [inputDictionary objectForKey:HKSampleKeyMetadata];
+    if (metadata == nil)
+        metadata = @{};
+    return [self getHKCorrelationWithStartDate:startDate endDate:endDate correlationTypeString:correlationTypeString objects:objects metadata:metadata error:error];
+}
+
+//Helper to isolate error checking on inputs for plugin
+-(BOOL) inputDictionary:(NSDictionary*) inputDictionary hasRequiredKey:(NSString*) key error:(NSError**) error {
+    if ([inputDictionary objectForKey:HKSampleKeyStartDate] == nil){
```
Comment (EddyVerbruggen) [inline src/ios/HealthKit.m]: Will change this to: `objectForKey:key`

**Comment by EddyVerbruggen** [comment]
:+1: Thanks for your great work!

I've tested your code and provided a few samples so ppl will know how to use these new functions. It's available in 0.3.4.

**Comment by paulmillr** [comment]
Thanks Eddy!

**Pull request merged**